joomla/joomla-cms
# [6.0] Languagefilter: Refactoring of routing

Pull Request for issue #24315, #25449, #39181

### Summary of Changes
This is a refactoring of the routing component of the languagefilter plugin. The goal are as follows:
- Allow to parse URLs more than once. Right now the parseing of the languagefilter plugin sets a lot of state and even does redirects.
- When the prefix of the default language is set to be removed, the redirect for the language switch is removed.
- Make the code simpler and more reliable.

~~Right now this PR is a Proof Of Concept and is NOT ready. This is to collect feedback from the community and other developers.~~

### Testing Instructions
Setup different multilanguage sites and apply the patch. The default behavior should not change, except that when the default prefix is removed, you can switch the language without an additional redirect. 



### Link to documentations
Please select:
